src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx

```diff
@@ -0,0 +1,68 @@
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { TIERS } from 'game/TIERS';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import SPELLS from 'common/SPELLS';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import Events, { DamageEvent } from 'parser/core/Events';
+import { calculateEffectiveDamageFromCritIncrease } from 'parser/core/EventCalculateLib';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import { formatNumber } from 'common/format';
+
+export default class T29MMTier4P extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+  };
+
+  totalDamage: number = 0;
+  totalProcs: number = 0;
+
+  protected statTracker!: StatTracker;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.has4PieceByTier(TIERS.T29);
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.ARCANE_SHOT),
+      this.onShotDamage,
+    );
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.MULTISHOT_MM),
+      this.onShotDamage,
+    );
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.FOCUSING_AIM), () => {
+      this.totalProcs += 1;
+    });
+  }
+
+  onShotDamage(event: DamageEvent) {
+    if (!this.hasFocusingAim(event)) {
+      return;
+    }
+    const effectiveDamgeFromCritIncrease = calculateEffectiveDamageFromCritIncrease(
+      event,
+      this.statTracker.currentCritPercentage,
+      1,
```

Comment:
I'd probably put the 1 in a constant (could be in this file), so we know what it represents if we ever go back and look at this.

Reply:
Extracted it to a constant 👍